### PR TITLE
Update telemetry metrics units and unit test

### PR DESCRIPTION
**Pull request by lkomali**
This is a small PR to update the telemetry metrics units and related unit test. 
Also added a data property to telemetry metrics for accessibility.

### CODE REVIEW

File: genai-perf/genai_perf/metrics/telemetry_metrics.py [hunk 1]
```diff
@@ -39,12 +39,12 @@ class TelemetryMetrics:
...
     TELEMETRY_METRICS = [
-        MetricMetadata("gpu_power_usage", "watts"),
-        MetricMetadata("gpu_power_limit", "watts"),
-        MetricMetadata("energy_consumption", "joules"),
-        MetricMetadata("gpu_utilization", "percentage"),
-        MetricMetadata("total_gpu_memory", "bytes"),
-        MetricMetadata("gpu_memory_used", "bytes"),
+        MetricMetadata("gpu_power_usage", "W"),
+        MetricMetadata("gpu_power_limit", "W"),
+        MetricMetadata("energy_consumption", "MJ"),
```
Comment (dyastremsky) [inline genai-perf/genai_perf/metrics/telemetry_metrics.py:44]: Is this joules or megajoules? Looks like the units changed.

File: genai-perf/genai_perf/metrics/telemetry_metrics.py [hunk 2]
```diff
@@ -80,3 +80,8 @@ def __repr__(self):
     @property
     def telemetry_metrics(self) -> List[MetricMetadata]:
         return self.TELEMETRY_METRICS
+
+    @property
+    def data(self) -> dict:
```
Comment (dyastremsky) [inline genai-perf/genai_perf/metrics/telemetry_metrics.py:85]: Will this get used anywhere?

**Review by dyastremsky — COMMENTED** [review]
Great work! Commented two questions.

File: genai-perf/genai_perf/metrics/telemetry_metrics.py (hunk 1 above)
Comment (lkomali) [reply genai-perf/genai_perf/metrics/telemetry_metrics.py:44]: yeah, it's MJ. The energy_consumption, and memory related metrics will be scaled for readability.
The scaling part will be in TelemetryStatistics class just like how scaling is done right not in statistics class.

File: genai-perf/genai_perf/metrics/telemetry_metrics.py (hunk 2 above)
Comment (lkomali) [reply genai-perf/genai_perf/metrics/telemetry_metrics.py:85]: This will be used in TelemetryStatistics.

**Review by dyastremsky — APPROVED** [review]

**Review by nv-hwoo — APPROVED** [review]

**Pull request merged**